core/utils.go

```diff
@@ -86,3 +87,9 @@ func ConvertBytesToQuorumNumbers(input []byte) []eigentypes.QuorumNum {
 	}
 	return output
 }
+
+type Clock struct {
+	Now func() time.Time
+}
+
+var SystemClock = &Clock{Now: time.Now}
```

Comment:
Why have this as a pointer?

Reply:
I think it's a leftover of my first approach which used interfaces. Changing it to a value.

Reply:
Didn't mind it too much at this point, but any specific reason you changed the approach?

Reply:
Less code honestly. If we were to use an interface we would have:

```go
type Clock interface {
    Now() time.Time
}

type SystemClock struct { }
func (sc *SystemClock) Now() time.Time { return time.Now() }
var _ Clock = &SystemClock{}

type FixedClock struct {
    NowValue time.Time
}

func (fc *FixedClock) Now() time.Time { return fc.NowValue }
var _ Clock = &FixedClock{}
```

You really don't win anything over a `struct` that holds a function (final approach). If we wanted to use something more configurable I suggest looking for a proper "clock" library (ex. https://github.com/kubernetes/utils/blob/master/clock/clock.go)

Reply:
Yup, that's okay for now